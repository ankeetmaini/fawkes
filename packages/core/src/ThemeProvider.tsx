import React, { FunctionComponent } from 'react';
import { Color } from '../types/index';
import Theme from './index';

interface ProviderProps {
  customColorScheme?: Color;
}

const { colorScheme, createTypography } = Theme;
const initialContext = {
  colorScheme,
  typography: createTypography(colorScheme),
};

const ThemeContext = React.createContext(initialContext);

const ThemeProvider: FunctionComponent<ProviderProps> = ({
  customColorScheme,
  children,
}) => {
  const themeContext = customColorScheme
    ? {
        colorScheme: customColorScheme,
        typography: createTypography(customColorScheme),
      }
    : initialContext;
  return (
    <ThemeContext.Provider value={themeContext}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;
