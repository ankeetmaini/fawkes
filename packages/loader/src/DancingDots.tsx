import React, { FunctionComponent } from 'react';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';

const dance = keyframes`
  0%, 55%, 100% {
    transform: translate3d(0, 0, 0);
  }
  60% {
    transform: translate3d(0, -8px, 0);
  }
  80% {
    transform: translate3d(0, 4px, 0);
  }
  90% {
    transform: translate3d(0, -3px, 0);
  }
  95% {
    transform: translate3d(0, 1px, 0);
  }
`;

const DancingDots: FunctionComponent<{ dotCount: number }> = ({ dotCount }) => {
  let delay = 0;
  return (
    <>
      {Array.from({ length: dotCount })
        .map((_, i) => i + 1)
        .map(i => {
          const StyledSpan = styled.span`
            display: inline-block;
            margin: 2px;
            animation: ${dance} 1800ms linear infinite
              ${i === 1 ? 0 : i % 2 === 0 ? (delay += 100) : (delay += 200)}ms;
          `;
          return <StyledSpan key={i}>.</StyledSpan>;
        })}
    </>
  );
};

export default DancingDots;
