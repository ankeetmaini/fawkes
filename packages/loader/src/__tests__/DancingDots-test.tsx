import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ThemeProvider from '@fawkes/core';
import { matchers } from 'jest-emotion';
import DancingDots from '../DancingDots';

expect.extend(matchers);
afterEach(cleanup);

describe('loaders', () => {
  test('dancing dots renders', async () => {
    const { asFragment } = render(
      <ThemeProvider>
        <DancingDots dotCount={3} />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('dancing dots animation delay is correct', () => {
    const { container } = render(<DancingDots dotCount={3} />);
    const spans = container.querySelectorAll('span');
    expect(spans[0]).toHaveStyleRule(
      'animation',
      'animation-0 1800ms linear infinite 0ms',
    );
    expect(spans[1]).toHaveStyleRule(
      'animation',
      'animation-0 1800ms linear infinite 100ms',
    );
    expect(spans[2]).toHaveStyleRule(
      'animation',
      'animation-0 1800ms linear infinite 300ms',
    );
  });
});
