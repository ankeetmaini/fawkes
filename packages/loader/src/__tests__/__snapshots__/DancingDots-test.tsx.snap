// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loaders dancing dots renders 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

@keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

@keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

.emotion-0 {
  display: inline-block;
  margin: 2px;
  -webkit-animation: animation-0 1800ms linear infinite 0ms;
  animation: animation-0 1800ms linear infinite 0ms;
}

<span
    class="emotion-0"
  >
    .
  </span>
  @keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

@keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

@keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

.emotion-0 {
  display: inline-block;
  margin: 2px;
  -webkit-animation: animation-0 1800ms linear infinite 100ms;
  animation: animation-0 1800ms linear infinite 100ms;
}

<span
    class="emotion-0"
  >
    .
  </span>
  @keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

@keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

@keyframes animation-0 {
  0%, 55%, 100% {
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  60% {
    -webkit-transform: translate3d(0,-8px,0);
    -ms-transform: translate3d(0,-8px,0);
    transform: translate3d(0,-8px,0);
  }

  80% {
    -webkit-transform: translate3d(0,4px,0);
    -ms-transform: translate3d(0,4px,0);
    transform: translate3d(0,4px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-3px,0);
    -ms-transform: translate3d(0,-3px,0);
    transform: translate3d(0,-3px,0);
  }

  95% {
    -webkit-transform: translate3d(0,1px,0);
    -ms-transform: translate3d(0,1px,0);
    transform: translate3d(0,1px,0);
  }
}

.emotion-0 {
  display: inline-block;
  margin: 2px;
  -webkit-animation: animation-0 1800ms linear infinite 300ms;
  animation: animation-0 1800ms linear infinite 300ms;
}

<span
    class="emotion-0"
  >
    .
  </span>
</DocumentFragment>
`;
